core/src/epicli/cli/engine/terraform/TerraformRunner.py

```diff
@@ -52,22 +52,27 @@ def azure_login(self):
                 # Create the service principal, for now we use the default subscription
                 self.logger.info('Creating service principal')
                 cluster_name = self.cluster_model.specification.name.lower()
-                cluster_prefix = self.cluster_model.specification.prefix.lower()               
+                cluster_prefix = self.cluster_model.specification.prefix.lower()
                 resource_group_name = resource_name(cluster_prefix, cluster_name, 'rg')
                 sp = apiproxy.create_sp(resource_group_name, subscription['id'])
                 sp['subscriptionId'] = subscription['id']
                 save_sp(sp, self.cluster_model.specification.name)
             else:
-                self.logger.info('Using service principal from file')
+                self.logger.info(f'Using service principal from file {sp_file}')
                 sp = load_yaml_file(sp_file)
 
             # Login as SP and get the default subscription.
             subscription = apiproxy.login_sp(sp)
 
             if 'subscriptionId' in sp:
-                # Set active subscription if sp contains it. 
+                # Set active subscription if sp contains it.
+                self.logger.debug(f"subscriptionId from sp file: {sp['subscriptionId']}")
                 apiproxy.set_active_subscribtion(sp['subscriptionId'])
-                self.new_env['ARM_SUBSCRIPTION_ID'] = sp['subscriptionId']
+                arm_subscription_id = sp['subscriptionId']
+                if "/" in sp['subscriptionId']:
+                    self.logger.debug(f"WARN Slash detected in the subscription name {sp['subscriptionId']}. Will parse the ID and use it instead")
+                    arm_subscription_id = next((_sub['id'] for _sub in subscription if _sub['name'] == sp['subscriptionId']), None)
```

Comment:
AFAIU you are fixing incorrectly created service principal file. I assume that service principal file you are using has `subscriptionId` and `name` fields filled with name.

Reply:
Please review the existing code as I don't have much time on this, but you will see we accept two types of ID, one UUID and the other format. 

It is just takes it too long a simple change like this with all test and uses cases presented. Sad!

Reply:
Sorry for inconvenience and late notice but this PR just looks incorrect. Putting `name` into `subscriptionId` field looks like misuse. Service principal file is documented [here](https://github.com/epiphany-platform/epiphany/blob/88cfb9e559f2cdd20c39d2d072d6436f4bed5c39/docs/home/howto/CLUSTER.md#how-to-create-an-epiphany-cluster-on-a-cloud-provider) in point 3: 

> Where the clustername is the name you specified under specification.name in your cluster yaml. Then in the terraform folder add the file named sp.yml and fill it with the service priciple information like so:
> 
> ```yaml
> appId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
> displayName: "app-name"
> name: "http://app-name"
> password: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
> tenant: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
> subscriptionId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx"
> ```

From PR correctness point of view, we tested this change and if we put name without `/` in it, this change will not work, so even if I didn't mind misusing field `if "/" in sp['subscriptionId']` is incorrect and some "validate if is UUID" should be placed here instead.

Reply:
It is up to the team. At the time in ABB RND team, we hit the problems, our sp.yaml is correct (in terms of correctness) and the problems is describe in this pr, and has been fixed by using this change PR.

That is the fact. If you don't ACKed it then when the RND team move to new version they will hit it again (if they do not use my custom build image/branch which has the fix)

I no longer work for ABB thus I do not have time to debug anymore. From my memory by tracing the code, epiphany accept two format, the UUID or the name - and when the name has slash the azure behavior makes it errors. Read again about my description.

I do it with best of considerations and judgement at the time and I retain. If problem goes away then it might be Azure has fixed their issues (that is if they allow slash in name then in API request they would interpret the encode of slash %F02 to be a string rather than a API path separator).